### PR TITLE
Feature/shutter control

### PR DESCRIPTION
**Pull request by coeing**
TODO:
- [x] Stopping not working as implemented, check if possible or remove
- [x] Add PercentType command to set precise percentage of shutter level
- [x] Move service name "ShutterControl" into constant variable
- [x] Remove Gateway-ID from header of putState request

### CODE REVIEW

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschSHCBridgeHandler.java [hunk 1]
```diff
@@ -537,9 +540,39 @@ private Boolean getRooms() {
...
+            return;
+        }
+
+        // Create request
+        String url = "https://" + config.ipAddress + ":8444/smarthome/devices/" + deviceId + "/services/" + serviceName
+                + "/state";
+        Gson gson = new Gson();
+        String body = gson.toJson(state);
+        Request request = this.httpClient.newRequest(url).header("Content-Type", "application/json")
+                .header("Accept", "application/json").header("Gateway-ID", "64-DA-A0-02-14-9B").method(PUT)
```
Comment (stefan-kaestle) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschSHCBridgeHandler.java:561]: There is a hard-coded MAC address in here :-)
Not sure if that's a problem?

Comment (stefan-kaestle) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschSHCBridgeHandler.java:561]: I think I have them in various places too :-P

File: bundles/org.openhab.binding.boschshc/.classpath [hunk 2]
```diff
@@ -28,5 +28,19 @@
 			<attribute name="maven.pomderived" value="true"/>
 		</attributes>
 	</classpathentry>
+	<classpathentry kind="src" output="target/test-classes" path="target/generated-test-sources/test-annotations">
```
Comment (stefan-kaestle) [inline bundles/org.openhab.binding.boschshc/.classpath:31]: Not sure what this thing does? :-)

**Comment by coeing** [comment]
Stopping the shutters from moving doesn't seem possible with the current API, so I created an issue for it: https://github.com/BoschSmartHome/bosch-shc-api-docs/issues/34 

Maybe I can find a workaround in the meantime.

**Comment by coeing** [comment]
Alright, this was a short but very successful coding session :) I handled all the remaining TODOs, so from my side the pull request is ready to merge. But please check if it is okay from your side as well!

I am using some general functions that can also be used in other places, but I would like to do the refactoring in a separate branch.

Next time I have time I will add some issues for the parts I'd like to change, so we can discuss them if necessary.

**Comment by coeing** [comment]
@stefan-kaestle Thanks for the review :) 

About your points:
- I can add some information about the new supported thing to the Readme.md, that's a good point!
- I would suggest to have a list of contributors in the Readme.md and not in every single code file. If anybody is interested who created which code fragment, he/she could easily use Git Blame for it :) 

If you give me write access to the repository, I'll merge my branch after the addition of the shutter control info to the Readme.md. Otherwise I'll let you know when I did the addition :)

**Comment by coeing** [comment]
Alright, I added the Shutter Control device to the supported device section of the Readme.md and added a null reference check because on initialization the device state doesn't seems to be fetched correctly.

Will merge the pull request now! :scream:

**Comment by stefan-kaestle** [comment]
Great work! Thanks Christian! :tada::tada::tada::tada:

**Pull request merged**